### ISSUE TITLE
Set user-agent when interacting with CCloud

### ISSUE THREAD

**Issue by jlrobins** [confluentinc/ide-sidecar#182]
Would be good for sidecar to interact with CCloud using a definite APIF team-approved User-Agent value that they will know how to aggregate.

**Comment by rhauch** [comment]
The Confluent for VS Code extension should include the following User-Agent header value in CCloud requests:
```
Confluent-for-VSCode/v0.1.2 (https://confluent.io; support@confluent.io) sidecar/v0.10.20 (linux/amd64)
```

Changes to CCloud to recognize this agent are under way and should be in production in a few weeks, though those changes are not required and should not dictate the timing of this work.

**Comment by Cerchie** [comment]
addressed via #202
